src/libraries/Microsoft.PowerFx.Json/Functions/JsonUntypedObject.cs

```diff
@@ -103,6 +103,12 @@ public bool TryGetProperty(string value, out IUntypedObject result)
             var res = _element.TryGetProperty(value, out var je);
             result = new JsonUntypedObject(je);
             return res;
+        }
+
+        public bool TryGetPropertyNames(out IEnumerable<string> result)
+        {
+            result = _element.EnumerateObject().Select(x => x.Name);

```

Comment:
>EnumerateObject [](http://example.com/codeflow?start=30&length=15)

While this is consistent with the rest of the code, we don't check if ValueKind is Object

Reply:
Added the check